safe_transaction_service/history/serializers.py

```
@@ -363,91 +415,101 @@ def save(self, **kwargs):
-            raise ValidationError(
-                "More than one signatures detected, just one is expected"
+    def validate_delegator_signature(
+        self,
+        delegate: ChecksumAddress,
+        signature: EthereumBytes,
+        signer: ChecksumAddress,
+    ) -> bool:
+        ethereum_client = get_auto_ethereum_client()
+        chain_id = ethereum_client.get_chain_id()
+        # Accept a message with the current topt and the previous totp (to prevent replay attacks)
+        for previous_totp, chain_id in list(
```

Ruff (0.8.2)

Loop control variable `chain_id` overrides iterable it iterates

(B020)